cmd/scale.go

```diff
@@ -404,6 +414,29 @@ func (sc *scaleCmd) run(cmd *cobra.Command, args []string) error {
 	return sc.saveAPIModel()
 }
 
+func (sc *scaleCmd) setFaultDomainCount(ctx context.Context, resourceGroup string, vmasIds []string) error {
+	count := engine.PlatformFaultDomainCountNotSet
+
+	for _, id := range vmasIds {
+		// extract the last element of the id for VMAS name
+		ss := strings.Split(id, "/")
+		name := ss[len(ss)-1]
+		vmas, err := sc.client.GetAvailabilitySet(ctx, resourceGroup, name)
+		if err != nil {
+			return err
+		}
+		// TODO: return error if all VMASes don't agree on this value?
```

Comment:
An optimization here is to return as soon as one availability set's `platformFaultDomainCount` is found, on the assumption that all availability sets in the cluster will share that value. This should hold true for any cluster created with `aks-engine` or AKS, and it's not possible to change the value after creation (or this PR would have been quite a bit simpler).

My first approach was to build a map of `vmasName` -> `platformFaultDomainCount` so I could update each VMAS resource in the template specifically. But the names are referenced only very indirectly at template generation time, so the association could not easily be made, and in practice the values could always boil down to a single number.

But it's worth asking if this is a short-sighted optimization.